helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java

```diff
@@ -274,59 +276,208 @@ public ResourceAssignment computeBestPossiblePartitionState(ResourceControllerDa
     return partitionMapping;
   }
 
+
   @Override
   protected Map<String, String> computeBestPossibleStateForPartition(Set<String> liveInstances,
```

Comment:
Since this is not public API, can we directly change liveInstances to cache? And refactor the abstract rebalancer part?

If not, I would suggest to add cache as the last argument to form a new function. Then we can have old function call pass null value to cache to skip the WAGED check. That can reduce more redundant code.

Reply:
I would love it if i can change this protected API directly.  it is the right fix as well. Please let me know as this will simplify code a lot.

Reply:
My bad. This method cannot be changed because AbstractRebalancer has been extended by user. Change method signature would lead build break...

Reply:
thanks for your suggetions. i have re-organized the code and it looks much much better. thanks